src/DssSpell.sol

```diff
@@ -23,47 +23,45 @@ import "dss-exec-lib/DssAction.sol";
 }
 
+
 contract DssSpellAction is DssAction, DssSpellCollateralOnboardingAction {
     // Provides a descriptive tag for bot consumption
     // This should be modified weekly to provide a summary of the actions
-    // Hash: seth keccak -- "$(wget https://raw.githubusercontent.com/makerdao/community/3224f50b0b5a9301831213ed858bc1d206de8e40/governance/votes/Executive%20vote%20-%20December%2010%2C%202021.md -q -O - 2>/dev/null)"
+    // Hash: seth keccak -- "$(wget https://raw.githubusercontent.com/makerdao/community/45ce05217966639a20dda7334f75e2c6ac0c69af/governance/votes/Executive%20vote%20-%20January%2014%2C%202022.md -q -O - 2>/dev/null)"
     string public constant override description =
-        "2021-12-10 MakerDAO Executive Spell | Hash: 0x9cc240b4c1027d3dc1abb24ecc598703352d8135c8a1670a6591e9a334836e5d";
+        "2022-01-14 MakerDAO Executive Spell | Hash: 0x648463c383a85878ec3db7a15296baa321a881b380579879735704b5cf998a9a";
 
     // --- Math ---
-    uint256 constant MILLION = 10**6;
+    uint256 constant MILLION = 10 ** 6;
+    uint256 constant WAD     = 10 ** 18;
 
     // --- Ilks ---
-    bytes32 constant WSTETH_A = "WSTETH-A";
-    bytes32 constant MATIC_A  = "MATIC-A";
+    bytes32 constant UNIV2WBTCETH_A = "UNIV2WBTCETH-A";
+    bytes32 constant UNIV2UNIETH_A  = "UNIV2UNIETH-A";
+    bytes32 constant UNIV2DAIETH_A  = "UNIV2DAIETH-A";
+    bytes32 constant UNIV2USDCETH_A = "UNIV2USDCETH-A";
+    bytes32 constant UNIV2WBTCDAI_A = "UNIV2WBTCDAI-A";
 
     // --- Wallet addresses ---
-    address constant GRO_WALLET = 0x7800C137A645c07132886539217ce192b9F0528e;
-    address constant ORA_WALLET = 0x2d09B7b95f3F312ba6dDfB77bA6971786c5b50Cf;
-    address constant PE_WALLET  = 0xe2c16c308b843eD02B09156388Cb240cEd58C01c;
-
-    // --- Dates ---
-    uint256 constant MAY_01_2021 = 1619827200;
-    uint256 constant JUN_21_2021 = 1624233600;
-    uint256 constant JUL_01_2021 = 1625097600;
-    uint256 constant SEP_13_2021 = 1631491200;
-    uint256 constant SEP_20_2021 = 1632096000;
+    address constant FLIP_FLOP_FLAP_WALLET  = 0x688d508f3a6B0a377e266405A1583B3316f9A2B3;
+    address constant FEEDBLACK_LOOPS_WALLET = 0x80882f2A36d49fC46C3c654F7f9cB9a2Bf0423e1;
+    address constant SCHUPPI_WALLET         = 0x89C5d54C979f682F40b73a9FC39F338C88B434c6;
+    address constant MAKERMAN_WALLET        = 0x9AC6A6B24bCd789Fa59A175c0514f33255e1e6D0;
+    address constant ACRE_INVEST_WALLET     = 0x5b9C98e8A3D9Db6cd4B4B4C1F92D0A551D06F00D;
+    address constant JUSTIN_CASE_WALLET     = 0xE070c2dCfcf6C6409202A8a210f71D51dbAe9473;
+    address constant GFX_LABS_WALLET        = 0xa6e8772af29b29B9202a073f8E36f447689BEef6;
+
+    // --- Optimism Recovery Mainnet Addresses
+    address immutable MCD_DAI             = DssExecLib.dai();
+    address immutable OPTIMISM_ESCROW     = DssExecLib.getChangelogAddress("OPTIMISM_ESCROW");
+    address constant LOST_SOME_DAI_WALLET = 0xc9b48B787141595156d9a7aca4BC7De1Ca7b5eF6;
```

Comment:
I just want to call out how dangerous this action can be if we don't do this all in one transaction.  Let's make sure we always test that the approval is 0 after, which we do.

Reply:
yeah... I was thinking it was good assert that before and after too. though I'm not sure I see the danger you're talking about, if gov wanted to "exploit" a lingering approval, they would need to use a spell and if they are using a spell they can just set the approval again.  or are you thinking for a point in the future where gov has revoked permission to call approve and therefore the lingering approval would be a perms leak effectively?

Reply:
Also, validated the optimism transaction to `burn()` was from `0xc9b48B787141595156d9a7aca4BC7De1Ca7b5eF6` and the request in the forums came from the same person with the signed message.